src/fileio/dataformat.cxx

```diff
@@ -48,7 +48,7 @@ void DataFormat::writeFieldAttributes(const std::string& name, const FieldPerp&
   int yindex = f.getIndex();
   if (yindex >= 0 and yindex < fieldmesh.LocalNy) {
     // write global y-index as attribute
-    setAttribute(name, "yindex_global", fieldmesh.getGlobalYIndex(f.getIndex()));
+    setAttribute(name, "yindex_global", fieldmesh.getGlobalYIndexNoBoundaries(f.getIndex()));
```

Comment:
There is a potential problem with this fix, that someone might save two `FieldPerp`s on different processors with the same `getGlobalYIndex()` - i.e. at the inner and outer upper targets in a double-null topology. This is actually what I am using the `FieldPerp` support in restarts for, and because the sheath boundary condition is saved at a `CELL_YLOW` position, it is exactly the same global index, if we use `getGlobalYIndexNoBoundaries()`. If the outputs are saved into a single file, I'm not sure what would happen but probably we would only get one of the `FieldPerp`s saved.

I think the answer is we need `getLocalYIndex(int yglobal)` and `getLocalYIndexNoBoundaries(int yglobal_no_boundaries)` methods (sorry for adding more stuff to `Mesh`, but we can deprecate `YLOCAL()` in favour of `getLocalYIndexNoBoundaries()` so it's only one more small method!). Then we use `getLocalYIndex(yindex_global)` below https://github.com/boutproject/BOUT-dev/blob/e926569e23a634fe6b363cff30a9b7a0e70c8753/src/fileio/dataformat.cxx#L82

Reply:
I thought something like that might be necessary! Ok, I'll add those two methods